RestaurantReservation.DB/Repositories/RestaurantsRepository.cs

```diff
@@ -0,0 +1,69 @@
+using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.DB.EFContext;
+using RestaurantReservation.DB.Exceptions;
+using RestaurantReservation.DB.Models;
+
+namespace RestaurantReservation.DB.Repositories;
+
+public class RestaurantsRepository
+{
+    private readonly RestaurantContext _restaurantContext;
+
+    public RestaurantsRepository(RestaurantContext restaurantContext)
+    {
+        _restaurantContext = restaurantContext;
+    }
+    
+    public async Task<decimal> CalculateRestaurantRevenue(int restaurantId)
+    {
+        return (decimal) await _restaurantContext.Database.ExecuteSqlRawAsync(
+            """
+                DECLARE @revenue DECIMAL(12, 4);
+                EXEC @revenue = dbo.fn_RestaurantTotalRevenue @restaurant_id = {0};
+                
+                SELECT @revenue
+            """,
+            restaurantId
+        );
+    }
+    
+    public async Task AddNewRestaurant(Restaurant restaurant)
+    {
+        if (!_restaurantContext.Restaurants.Any(r => r.PhoneNumber == restaurant.PhoneNumber))
```

Comment:
so i can have 2 restaurant with the same name but different Phone Number ? update you condition to more proper condition

Reply:
It can't happen that two parties have the same phone number, phone numbers are unique to the person holding that number.